h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -139,7 +144,17 @@
     _read = true;               // Reading by default
     _firstPage = true;
     // Read Inet from socket, port from the stream, figure out H2ONode
-    _h2o = H2ONode.intern(remoteAddress, getPort());
+    if(remoteAddress!=null) {
+      _h2o = H2ONode.intern(remoteAddress, getPort());
+    }else{
+      // In case the communication originates from non-h2o node, we set _h2o node to null.
+      // It is done for 2 reasons:
+      //  - H2ONode.intern creates a new thread and if there's a lot of connections
+      //    from non-h2o environment, it could end up with too many open files exception.
+      //  - H2OIntern also reads port (getPort()) and additional information which we do not send
+      //    in communication originating from non-h2o nodes
+      _h2o = null;
```

Comment:
yes, the asserts are there, but `_h2o = null` is set only in the case where we communicate only using TCP TYPE 3 - external, which is handled completely by `ExternalFrameHandler` thus not affecting the rest of code at all.

Technically, It is true that it could be replaced by `ByteBuffer`, however I don't see any added value since I would just have to create a lot of duplicate methods for sending data over the network - exactly what `AutoBuffer` does right now.

So from my point of view this is ok, tests are passing and it doesn't affect previous code at all.